### PR TITLE
Pass in JAVA_IMPL value to test compile process

### PR DESCRIPTION
**Pull request by sophia-guo**
Pass in JAVA_IMPL value to test compile process, which could be consumed
to select different sdk version as this test framework is going to a
general framework to test different sdk.

[ci skip]

Signed-off-by: Sophia Guo <sophiag@ca.ibm.com>

### CODE REVIEW

**Comment by sophia-guo** [comment]
@llxia

File: test/TestConfig/makefile [hunk 1]
```diff
@@ -69,4 +69,4 @@ getdependency:
 # compile all tests under $(TEST_ROOT)
 #######################################
 compile: getdependency
-	ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJAVA_BIN=$(JAVA_BIN) -DJAVA_VERSION=$(JAVA_VERSION) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DJCK_ROOT=$(JCK_ROOT)
+	ant -f scripts$(D)build_test.xml -DTEST_ROOT=$(TEST_ROOT) -DBUILD_ROOT=$(BUILD_ROOT) -DJAVA_BIN=$(JAVA_BIN) -DJAVA_VERSION=$(JAVA_VERSION) -DJCL_VERSION=$(JCL_VERSION) -DBUILD_LIST=${BUILD_LIST} -DRESOURCES_DIR=${RESOURCES_DIR} -DJAVA_IMPL=${JAVA_IMPL} -DJCK_ROOT=$(JCK_ROOT)
```
Comment (llxia) [inline test/TestConfig/makefile]: Do you mind of changing { } to ( ) for all passing values?

File: test/TestConfig/settings.mk [hunk 2]
```diff
@@ -211,6 +211,10 @@ else
 export JCL_VERSION:=$(JCL_VERSION)
 endif
 
+ifndef JAVA_IMPL
+export JAVA_IMPL:=openj9
+endif
+
```
Comment (llxia) [inline test/TestConfig/settings.mk]: Please move this code to the beginning of the file, after to JAVA_VERSION. Say line 60.
It is better keep all exports together in the beginning of the mk.
Please add `@$(ECHO) set JAVA_IMPL to $(JAVA_IMPL)` to setup_% target

Comment (sophia-guo) [reply test/TestConfig/settings.mk]: Updated.

File: test/TestConfig/makefile (hunk 1 above)
Comment (sophia-guo) [reply test/TestConfig/makefile]: Updated.

**Comment by llxia** [comment]
Jenkins test sanity plinux

**Comment by llxia** [comment]
Close this pull request as we have pull request for JVM_VERSION https://github.com/eclipse/openj9/pull/1154

**Pull request closed without merging**